packages/ember-glimmer/tests/integration/application/engine-test.js

```diff
@@ -572,4 +576,18 @@ moduleFor('Application test: engine rendering', class extends ApplicationTest {
       });
     });
   }
+
+  ['@test query params don\'t have stickiness by default between model'](assert) {
+    assert.expect(1);
+
+    this.setupAppAndRoutableEngine();
+    this.additionalEngineRegistrations(function() {
+      this.register('template:category', compile('{{#link-to "blog.category" 1337}}Category 1337{{/link-to}}'))
+    });
+
+    return this.visit('/blog/category/1?type=news').then(() => {
+      let href = this.element.querySelector('a').href;
+      assert.equal(href.match(/type=news/), null);
```

Comment:
I've updated this to use .endsWith since full href include port number. Let me know if you need further change.